client/src/data/selectors/userSelector.ts

```diff
@@ -2,13 +2,11 @@ import { createSelector } from 'reselect';
 import { userObject } from '../reducers/authReducers';
 
 const currentUser = (state: userObject) => {
-    console.log("HERE")
-    console.log(state)
-    return state.user && state.user.user ? state.user.user : null;
+    return state.user.user && state.user ? state.user.user : null;
```

Comment:
would it make sense to switch the order of `state.user.user`  and `state.user` in the conditional instead? we could use a conditional short circuit to make sure that `state.user` is not null before trying to read `state.user.user`